idaplugin/rematch/collectors/annotations/assembly.py

```diff
@@ -2,30 +2,60 @@
 import ida_funcs
 import ida_lines
 import idautils
+import ida_idaapi
+
+from math import log
+import re
 
 from . import annotation
 
 
 class AssemblyAnnotation(annotation.Annotation):
   type = 'assembly'
 
-  @classmethod
-  def data(cls, offset):
-    flow_chart = ida_gdl.FlowChart(ida_funcs.get_func(offset))
+  @staticmethod
+  def data(offset):
+    func = ida_funcs.get_func(offset)
+
+    def node_in_func(node):
+      if node.startEA < func.startEA:
+        return False
+      if node.startEA >= func.endEA:
+        return False
+      if node.endEA > func.endEA:
+        return False
+
+      return True
+
+    def clean(asm):
+      """This removes markers of function offsets, including hidden variable
+      length offsets that are of different length on 32 and 64 bit address IDA.
+      Otherwise, IDA of different offset lengths will truncate incorrect number
+      of bytes"""
+      hex_chars = int(log(ida_idaapi.BADADDR, 2) / 4)
+      pattern = r"\x01\([1-0a-zA-Z]{%s}([\w\s]*?)\x02%%\x02\)" % hex_chars
+      return re.sub(pattern, "\g<1>", asm)
+
+    # make sure only nodes inside the function are accounted for
+    # this solves cascaded functions (when multiple functions share same ends)
+    nodes = filter(node_in_func, ida_gdl.FlowChart(func))
+    node_ids = map(lambda n: n.id, nodes)
 
-    nodes = {}
-    for node in flow_chart:
-      assembly = [ida_lines.generate_disasm_line(ea)
+    nodes_data = []
+    for node in nodes:
+      assembly = [clean(ida_lines.generate_disasm_line(ea))
```

Comment:
> Why not use GetMnem , GetOpnd and GetOperandValue to get the line itself?

Color markers are exactly the reason I wanna use `ida_lines.generate_disasm_line`, because it gives you the disassembly line *with* the color markers.

> Does your filtering preserve the coloring?

Yes, it only remove one marker, which describes the offset of functions (probably for jump-to) functionality. We don't need it.